### ISSUE TITLE
CURL and CURL_OPTS not resetted between two calls

### ISSUE THREAD

**Issue by SylvainMarty** [hiboutik/hiboutikapi#8]
Hello there !

We are migrating your API client from `1.1.0` to `2.2.0` but we have some problems with this new version.
We are using a same instance of `HiboutikAPI` to call multiple time the Hiboutik API and the new client fail every time a POST query follow a PUT query.

Digging in the source code of `\Hiboutik\HiboutikAPI\HttpRequest` shows us that `$this->curl` and `$this->curl_opts` properties aren't resetted before each request which can lead to inconsistant call when the client is reused.

We are currently using the API v1 with basic authentication.

Here is a case of error when we use the new client like the old one : 
```php
$client = new \Hiboutik\HiboutikAPI('', '', '', '1');
// Put an update of a product
$client->putHiboutik("product/$id", [
    'product_attribute' => $key,
    'new_value' => $value
]);
// Post a new category
$client->postHiboutik('categories', [
    'category_name' => 'Test',
    'category_parent_id' => '0'
]);
```

Here is a dump of your client when the issue appears : 
```
Hiboutik\HiboutikAPI {#2633
  #hr: Hiboutik\HiboutikAPI\HttpRequest {#2762
    #curl: curl resource {@1465
      url: "https://guidapsylvain.hiboutik.com/apirest/categories"
      content_type: "application/json; charset=utf-8"
      http_code: 400
      header_size: 167
      request_size: 325
      filetime: -1
      ssl_verify_result: 0
      redirect_count: 0
      total_time: 0.066582
      namelookup_time: 3.8E-5
      connect_time: 0.020416
      pretransfer_time: 0.041819
      size_upload: 43.0
      size_download: 36.0
      speed_download: 540.0
      speed_upload: 645.0
      download_content_length: 36.0
      upload_content_length: 43.0
      starttransfer_time: 0.066296
      redirect_time: 0.0
      redirect_url: ""
      primary_ip: "92.222.49.32"
      certinfo: []
      primary_port: 443
      local_ip: "172.18.0.5"
      local_port: 50968
    }
    #current_headers: array:5 [
      "Date" => "Wed, 24 Oct 2018 08:24:47 GMT\r"
      "Server" => "Apache\r"
      "Content-Length" => "36\r"
      "Content-Type" => "application/json; charset=utf-8\r"
      "Connection" => "close\r"
    ]
    #send_headers: []
    #curl_opts: array:17 [
      58 => false
      75 => true
      74 => true
      19913 => true
      78 => 4
      13 => 10
      84 => 2
      10018 => "HiboutikAPI Client v2"
      20079 => Closure {#2761
        class: "Hiboutik\HiboutikAPI\HttpRequest"
        this: Hiboutik\HiboutikAPI\HttpRequest {#2762}
        parameters: {
          $ch: {}
          $header_data: {}
        }
        file: "/var/www/html/vendor/hiboutik/hiboutikapi/src/Hiboutik/HiboutikAPI/HttpRequest.php"
        line: "358 to 365"
      }
      107 => 1
      10005 => "sylvain@guidap.co:F74WL3KFAKJ590FCP651F9GYZXATG73L4Y8"
      10002 => "https://guidapsylvain.hiboutik.com/apirest/categories"
      80 => true
      10023 => []
      47 => true
      10015 => "category_name=Kitesurf&category_parent_id=0"
      10036 => "PUT"
    ]
  }
  #uri: "https://guidapsylvain.hiboutik.com/apirest"
  #request_type: ""
  #request_resource: ""
  #request_data: null
  +errorCode: 400
  +request_ok: false
}
```

**Comment by hiboutik** [comment]
Hello,

What is the PHP version on your server?
Have you checked the logs for an error like this one:
"PHP Fatal error:  Due to technical limitations of PHP <= 5.5.11 and HHVM, it is not possible to perform a post-redirect-get request using a php-curl-class Curl object that has already been used to perform other types of requests. Either use a new php-curl-class Curl object or upgrade your PHP engine."

Regards

**Comment by SylvainMarty** [comment]
Hi @hiboutik !
We are using PHP 7.0 and no error log were found in the logfiles.

To argue our hypothesis, you can see on the dump I posted yesterday that `#curl_opts` contains the option `47 => true` which says that its a POST request but it also contains `10036 => "PUT"` which is normally use to make a request with a custom method like `PUT`, `PATCH` or `DELETE`.

**Comment by hiboutik** [comment]
Hi, found the bug, it has been corrected in v2.3.0.
Thanks for input

**Comment by SylvainMarty** [comment]
Hi ! Thank you, I will test the fix in our stack on monday.
